content/tutorial-ios/ios-06.md

```diff
@@ -0,0 +1,193 @@
+
+
+### Using The `UpdatePokemonMutation` 
+
+In `PokemonDetailViewController.swift`, copy the following code snippet into `editAndSaveButtonPressed()` replacing the current line:
+
+```swift
+if isEditingPokemon {
+    // make sure required data is available
+    guard let name = nameTextField.text, name.characters.count > 0,
+        let url = imageURLTextField.text, url.characters.count > 0,
+        let id = pokemonDetails?.id else {
+            print(#function, "ERROR | Missing data")
+            return
+    }
+    
+    // start activity indicator
+    saveActivityIndicator.startAnimating()
+
+    // actually perform the mutation
+    let updatePokemonMutation = UpdatePokemonMutation(id: id, name: name, url: url)
+    apollo.perform(mutation: updatePokemonMutation) { [unowned self] result, error in
+        if let error = error {
+            print(#function, "ERROR | Could not update Pokemon: (\(error))")
+        }
+        else if let pokemonDetails = result?.data?.updatePokemon?.fragments.pokemonDetails {
+            self.pokemonDetails = pokemonDetails
+            self.saveActivityIndicator.stopAnimating()
+            self.isEditingPokemon = !self.isEditingPokemon
+        }
+    }
+}
+else {
+    isEditingPokemon = !isEditingPokemon
+}
+```
+
+That's a lot of of code, let's try to understand what it does! In the very beginning, we check if we're currently in the _editing state_ - if that is the case we are going to perform the mutation using the strings that are written in the text fields. Therefore, we first make sure that the required data is provided using a `guard` statement. We then start the activity indicator to indicate to the user that a network request has started. Finally, we instantiate the `UpdatePokemonMutation` that was generated by `apollo-codegen` based on the mutation we added in `PokemonDetailViewController.graphql`. Subsequently, we call `perform` on our `ApolloClient` instance passing the instance of the mutation as well as a callback to deal with the return value. Since we are using our `PokemonDetails` fragment again, we can simply extract this from the return value inside the callback and assign it the `pokemonDetails` property. This will automatically update the UI because of the `didSet` property observer. 
+
+Go ahead and test the new feature by modifying the name or image URL of one of the Pokemons in your Pokedex. You should see the UI being updated with the new values after a successful completion of the mutation. However, once again, if we navigate back to the `PokedexTableViewController`, the changes won't be reflected in the table view yet.
+
+
+### Updating The UI
+
+That is because the data in the `ownedPokemons` array in `PokedexTableViewController` is independent from the data we just updated in `PokemonDetailViewController`, so we have to manually update it. Let's do so again with a closure. Add the following property to `PokemonDetailViewController`:
```

Comment:
I think wording can be improved here. The data is not unrelated (it's supposed to be the same!) it's  just that Apollo doesn't know that - how would it?